engine/verification/assigner/blockconsumer/consumer_test.go

```diff
@@ -97,49 +97,6 @@ func TestProduceConsume(t *testing.T) {
 			unittest.RequireReturnsBefore(t, processAll.Wait, time.Second, "could not process all blocks on time")
 			unittest.RequireCloseBefore(t, consumer.Done(), time.Second, "could not terminate consumer")
 
-			// expects the mock engine receive all 10 blocks.
-			require.ElementsMatch(t, flow.GetIDs(blocks), flow.GetIDs(received))
-		})
-	})
-
-	// pushing 100 finalized blocks concurrently to block reader, with 3 workers on consumer and the processor finishes processing
-	// all blocks immediately, results in the processor receiving all 100 blocks:
-	// 100 blocks concurrently --> block reader --> consumer can read and push 3 blocks at a time to processor --> processor finishes blocks
-	// immediately.
-	t.Run("pushing 100 blocks concurrently, non-blocking, receives 100", func(t *testing.T) {
-		received := make([]*flow.Block, 0)
-		lock := &sync.Mutex{}
-		var processAll sync.WaitGroup
-
-		alwaysFinish := func(notifier module.ProcessingNotifier, block *flow.Block) {
-			lock.Lock()
-			defer lock.Unlock()
-
-			received = append(received, block)
-
-			go func() {
-				notifier.Notify(block.ID())
-				processAll.Done()
-			}()
-		}
-
-		withConsumer(t, 100, 3, alwaysFinish, func(consumer *BlockConsumer, blocks []*flow.Block) {
-			unittest.RequireCloseBefore(t, consumer.Ready(), time.Second, "could not start consumer")
-			processAll.Add(len(blocks))
-
-			for i := 0; i < len(blocks); i++ {
-				go func() {
```

Comment:
We don't need to call `OnFinalizedBlock` concurrently, because the follower engine never do that. The follower engine always call `OnFinalizedBlock` sequentially for each finalized block.

The reason this test case is flakey is that the `OnFinalizedBlock` actually triggers the consumer to check, and the consumer will ensure only one check will actually trigger the consumer to start workers to process the jobs concurrently. That being said, the concurrency is done at the consumer level. So the test will crash, if the workers have finished all the jobs, and has closed the database, but there is a goroutine here was still calling `OnFinalizedBock` which triggers the consumer to check again, which will visit the database while it has been closed, and causes the crash.

The fix is simply remove this test case, and replace by changing the previous `pushing 10 blocks, non-blocking, receives 10` tests to `pushing 100 blocks, non-blocking, receives 100`. This way, we still test the consumer processing 100 jobs concurrently